File: hackathon/security-copilot/frontend/src/types/comparison.ts
export interface RelatedSourceItem {
  title: string;
  url: string;
  summary: string;
  published_at: string | null;
  relevance_score: number;
  matched_terms: string[];
  is_topical_match: boolean;
}

export interface VendorCompareSource {
  source_id: string;
  name: string;
  site: string;
  tagline: string;
  description: string;
  feed_url: string;
  related: RelatedSourceItem | null;
}

export interface SourceCompareResponse {
  article_id: number;
  article_title: string;
  sources: VendorCompareSource[];
}

File: hackathon/security-copilot/backend/app/schemas/comparison.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class RelatedSourceItem(BaseModel):
    title: str
    url: str
    summary: str
    published_at: Optional[datetime] = None
    relevance_score: int
    matched_terms: list[str]
    is_topical_match: bool


class VendorCompareSource(BaseModel):
    source_id: str
    name: str
    site: str
    tagline: str
    description: str
    feed_url: str
    related: Optional[RelatedSourceItem] = None


class SourceCompareResponse(BaseModel):
    article_id: int
    article_title: str
    sources: list[VendorCompareSource]
